setup.py

```diff
@@ -69,6 +69,7 @@
         # listed as optional in tiktoken's pyproject.toml
         # (https://github.com/openai/tiktoken/blob/main/pyproject.toml#L9)
         "blobfile~=3.0.0",
+        "ray~=2.40",
```

Comment:
I believe this should be optional. In fact, instead of making it optional, we can just do a quick import check in the cluster implementation, and raise an error advising the use to install Ray themselves. For majority of use cases, people won't be using Ray. See [this](https://github.com/facebookresearch/fairseq2/blob/main/src/fairseq2/nn/_normalization.py#L21-L34) as an example.

Reply:
made ray as extra. also make sure the test and existing code works without install ray.